src/compiler/checker.ts

```diff
@@ -10338,17 +10345,8 @@ namespace ts {
             if (!inferredType) {
                 const inferences = getInferenceCandidates(context, index);
                 if (inferences.length) {
-                    // We widen inferred literal types if

```

Comment:
This code moves into `getInferenceCandidates` so that both type inference (this section) and type inference error reporting (the other call to `getInferenceCandidates`) widen the same way.